src/backend/mapServer.js

```diff
@@ -0,0 +1,69 @@
+const { promisify } = require("util");
+const getPort = require("get-port");
+const createServer = require("@mapeo/map-server").default;
+const log = require("debug")("map-server");
+
+const AsyncService = require("./upgrade-manager/async-service");
+
+// TODO: Does it make sense for this to extend AsyncService?
+class MapServer extends AsyncService {
+  /** @type {number?} */
+  #port;
+  /** @type {import('fastify').FastifyInstance} */
+  #fastify;
+
+  /** @type {boolean} */
+  #fastifyStarted = false;
+
+  /**
+   * @param {object} object
+   * @param {string} dbPath
+   *
+   */
+  constructor({ dbPath }) {
+    super();
+    this.#fastify = createServer({ logger: false }, { dbPath });
+  }
+
+  /**
+   * Start the server on the specified port. Listen on all interfaces.
+   *
+   * @returns {Promise<void>} Resolves with the port number when server is started
+   */
+  async _start() {
+    if (!this.#port) {
+      this.#port = await getPort();
+    }
```

Comment:
only get an available port on initial server startup. subsequent calls will reuse the port, which comes with a small risk